cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -79,10 +109,43 @@ class OpenAIPrompt(override val uid: String) extends Transformer
 
   def setSystemPrompt(value: String): this.type = set(systemPrompt, value)
 
+  val responseFormat = new Param[String](
+    this, "responseFormat", "The response format from the OpenAI API.")
+
+  def getResponseFormat: String = $(responseFormat)
+
+  def setResponseFormat(value: String): this.type = {
+    if (value.isEmpty) {
+      this
+    } else {
+      val normalizedValue = value.toLowerCase match {
+        case "json" => "json_object"
+        case other => other
+      }
+
+      // Validate the normalized value using the OpenAIResponseFormat enum
+      if (!OpenAIResponseFormat.values
+        .map(_.asInstanceOf[OpenAIResponseFormat.ResponseFormat].name)
+        .contains(normalizedValue)) {
+        throw new IllegalArgumentException("Response format must be valid for OpenAI API. " +
+          "Currently supported formats are " + OpenAIResponseFormat.values
+          .map(_.asInstanceOf[OpenAIResponseFormat.ResponseFormat].name)
+          .mkString(", "))
+      }
+
+      set(responseFormat, normalizedValue)
+    }
+  }
+
+  def setResponseFormat(value: OpenAIResponseFormat.ResponseFormat): this.type = {
+    this.setResponseFormat(value.name)
+  }
```

Comment:
this code seems duplicated, can it be abstracted and shared?

Reply:
I am now using shared code.